### PR TITLE
NI-SCOPE: removes fetch_measurement and read_measurement 

### PR DESCRIPTION
**Pull request by shilohbeeler-NI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).

- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.

~~- [ ] I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?

Removes fetch_measurement and read_measurement as they are redundant given the fetch_measurement_stats function, and we want there to be only one way to do any given task.

### What testing has been done?

Builds correctly and passes tests.

### CODE REVIEW

**Comment by sbethur** [comment]
ok to test

File: src/niscope/system_tests/test_system_niscope.py [hunk 1]
```diff
@@ -294,9 +261,9 @@ def test_waveform_processing(session):
     session.configure_horizontal_timing(50000000, test_record_length, 50.0, test_num_records, True)
     with session.initiate():
         session.add_waveform_processing(niscope.enums.ArrayMeasurement.DERIVATIVE)
-        processed_waveforms = session.channels[test_channels].fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
+        processed_waveforms = session.channels[test_channels]._fetch_measurement(niscope.enums.ScalarMeasurement.MID_REF_VOLTS, 5.0)
```
Comment (sbethur) [inline src/niscope/system_tests/test_system_niscope.py:264]: Not a good idea to use private methods in system tests. You can use `fetch_measurement_stats` instead, right?

File: src/niscope/system_tests/test_system_niscope.py [hunk 2]
```diff
@@ -261,18 +261,18 @@ def test_waveform_processing(session):
...
     assert len(processed_waveforms) == test_num_channels * test_num_records
     assert len(unprocessed_waveforms) == test_num_channels * test_num_records
     # Here the idea is to leave a large margin to not test too specifically for any returned values but to demonstrate that the waveform processing does
     # undeniably cause a consistent shift in the values returned. The "0" exception for processed is due to the nature of derivatives -- if two samples
     # next to each other are identical, the derivative will be 0.
     for processed, unprocessed in zip(processed_waveforms, unprocessed_waveforms):
-        assert abs(unprocessed) < 1
-        assert abs(processed) > 1 or processed == 0
+        assert abs(unprocessed.result) < 1
+        assert abs(processed.result) > 1 or processed == 0
```
Comment (sbethur) [inline src/niscope/system_tests/test_system_niscope.py:275]: `processed.result == 0`

Comment (shilohbeeler-NI) [reply src/niscope/system_tests/test_system_niscope.py:275]: Gotcha. Was just pushing so I could run the tests on my testing machine, thanks for the catch.

**Pull request merged**